providers/azure.go

```diff
@@ -283,3 +286,28 @@ func (p *AzureProvider) GetLoginURL(redirectURI, state string) string {
 func (p *AzureProvider) ValidateSession(ctx context.Context, s *sessions.SessionState) bool {
 	return validateToken(ctx, p, s.AccessToken, makeAzureHeader(s.AccessToken))
 }
+
+func (p *AzureProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
+	jwtParts := strings.Split(s.IDToken, ".")
```

Comment:
This will likely need to wait until this related PR for Azure wraps up: https://github.com/oauth2-proxy/oauth2-proxy/pull/914

You will need to take advantage of what that PR is now using to access the claims.

We don't trust any claims unless you verify the signature (There's a robust discussion in that PR on this stance if you want to take a peek).

There's new helper OIDC methods to facilitate this that PR is using to extract the `email` claim from the `id_token` -- you can likely take advantage of the same codepath.

Reply:
That makes sense. That pull request should cover almost everything I intend to do with mine, we just focused on different claims.

I did now pick the token verification and claim extraction from https://github.com/oauth2-proxy/oauth2-proxy/pull/914 into this PR and tested it with `oidc_issuer_url = "https://sts.windows.net/}tenantid}/"` in the config and this works fine.

After the other PR is merged I will check if my changes are still needed and if so rebase them and update this request